runtime/src/qio/qio_popen.c

```diff
@@ -447,7 +454,14 @@ qioerr qio_proc_communicate(
   while( do_input || do_output || do_error ) {
     if( do_input ) {
       err = _qio_channel_flush_qio_unlocked(input);
-      if( !err ) do_input = false;
+      if( !err ) {
+        qioerr err2 = 0;
+        do_input = false;
+        // Close input channel.
+        err = qio_channel_close(false, input);
+        err2 = qio_file_close(input_file);
+        if( err2 && !err ) err = err2;
```

Comment:
My thinking is that if a client of this function receives an error on closing one, it might try to respond by closing both, but if the other closed without error, would that generate a problem when closing something that was already closed?  Since we need to return something when both fail, the client of this function can't distinguish between a failure that means they need to try to close both again and a failure that means they should try closing just that one.  Are there assumptions that should be made here to avoid the "closing an already closed <whatever>" error?